mantidproject/mantid
# Diffraction loader for IN16B

**Description of work.**

This PR makes it possible to load diffraction data recorded on IN16B in Mantid. It thus consists in the new instrument geometry corresponding to these detectors, and an option in `LoadILLIndirect2` to actually load this data.

Note that it currently exists two concurrents nexus structure to store this diffraction data, which are both supported and tested. 

The instrument geometry is created via https://github.com/mantidproject/mantidgeometry/pull/173 . 

**To test:**
Test data can be found in `ExternalData/Testing/Data/UnitTest/ILL/IN16B/`
```
bats_ws = LoadILLIndirect(Filename='249290.nxs', OutputWorkspace="bats_ws", LoadDiffractionData=True)
print ("Number of histograms : {0}, expecting 2049.".format(bats_ws.getNumberHistograms()))
print ("Number of channels : {0}, expecting 2048.".format(bats_ws.blocksize()))

doppler_ws = LoadILLIndirect(Filename='276047.nxs', OutputWorkspace="doppler_ws", LoadDiffractionData=True)
print ("Number of histograms : {0}, expecting 2049.".format(doppler_ws.getNumberHistograms()))
print ("Number of channels : {0}, expecting 1024.".format(doppler_ws.blocksize()))
```
Open the instrument view, and check :
 - for the bats data, both the first and last tube are without non-zero values;
 - for the Doppler data, the tube the nearest to the beam line is only zeroes.



Closes #29096.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
